examples/dwarfdump.rs

```diff
@@ -472,32 +470,43 @@ fn main() {
         None
     };
 
-    let sup_mmap;
-    let sup_file = if let Some(sup_path) = matches.opt_str("sup") {
-        let file = match fs::File::open(&sup_path) {
+    let load_file = |path| {
+        let file = match fs::File::open(&path) {
             Ok(file) => file,
             Err(err) => {
-                eprintln!("Failed to open file '{}': {}", sup_path, err);
+                eprintln!("Failed to open file '{}': {}", path, err);
                 process::exit(1);
             }
         };
-        sup_mmap = match unsafe { memmap::Mmap::map(&file) } {
+        let mmap = match unsafe { memmap::Mmap::map(&file) } {
             Ok(mmap) => mmap,
             Err(err) => {
-                eprintln!("Failed to map file '{}': {}", sup_path, err);
+                eprintln!("Failed to map file '{}': {}", path, err);
                 process::exit(1);
             }
         };
-        match object::File::parse(&*sup_mmap) {
+        let mmap_ptr = mmap.as_ptr();
+        let mmap_len = mmap.len();
+        mem::forget(mmap);
+        match object::File::parse(unsafe { slice::from_raw_parts(mmap_ptr, mmap_len) }) {
```

Comment:
Ok, but an arena would be nicer.

Reply:
idk what this means in practice.

Reply:
See https://github.com/philipc/gimli/commit/e6e7557a1198983494617f984ebf803af2d660da